### PR TITLE
slightly decouple sql_json, queries, and results http endpoints from …

### PR DESCRIPTION
**Pull request by DiggidyDave**
- [ ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [X] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
Some of the views and view logic (http request handlers, template rendering, response construction, etc) is hyper-coupled to the business logic. In our case, we need to setup alternate endpoints with different internal auth, routes, etc to use the `sql_json`, `queries`, and `results` endpoints from `views.core.Superset`.  I think these should be more thoroughly decoupled (ie, views just doing view stuff, and calling into a completely separate library for business logic) but that is a lot more than I can chew off right now, so this tiny change is simply breaking those three http request handlers off so we other functions can call into the business logic from different endpoints.

Choose one##** BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by john-bodley** [comment]
@mistercrunch @villebro

File: superset/views/core.py [hunk 1]
```diff
@@ -2491,6 +2491,9 @@ def cache_key_exist(self, key):
     @expose("/results/<key>/")
     @event_logger.log_this
     def results(self, key):
+        return self.results_exec(key)
+
+    def results_exec(self, key):
```
Comment (villebro) [inline superset/views/core.py]: Type annotations here would be nice

File: superset/views/core.py [hunk 2]
```diff
@@ -2722,33 +2725,37 @@ def _sql_json_sync(
     @expose("/sql_json/", methods=["POST"])
     @event_logger.log_this
     def sql_json(self):
+        query_params = request.json
+        return self.sql_json_exec(query_params)
+
+    def sql_json_exec(self, query_params):
```
Comment (villebro) [inline superset/views/core.py]: Here, too

File: superset/views/core.py [hunk 3]
```diff
@@ -2722,33 +2725,37 @@ def _sql_json_sync(
...
         except json.decoder.JSONDecodeError:
             logging.warning(
-                f"Invalid template parameter {request.json.get('templateParams')}"
+                f"Invalid template parameter {query_params.get('templateParams')}"
                 " specified. Defaulting to empty dict"
             )
             template_params = {}
-        limit = request.json.get("queryLimit") or app.config["SQL_MAX_ROW"]
-        async_flag: bool = request.json.get("runAsync")
+        limit = query_params.get("queryLimit") or app.config["SQL_MAX_ROW"]
```
Comment (villebro) [inline superset/views/core.py]: `limit: int` I believe

File: superset/views/core.py [hunk 4]
```diff
@@ -2904,6 +2911,9 @@ def fetch_datasource_metadata(self):
     @has_access_api
     @expose("/queries/<last_updated_ms>")
     def queries(self, last_updated_ms):
+        return self.queries_exec(last_updated_ms)
+
+    def queries_exec(self, last_updated_us):
```
Comment (villebro) [inline superset/views/core.py]: Same here

**Review by villebro — COMMENTED** [review]
LGTM and makes sense, but would be great to add types to make sure we always add more rigidity to the codebase when touching up old code.

File: superset/views/core.py (hunk 1 above)
Comment (DiggidyDave) [reply superset/views/core.py]: type annotations added, good idea

**Comment by DiggidyDave** [comment]
@villebro added typing as suggested, thanks 👍

**Review by mistercrunch — APPROVED** [review]
This LGTM. I can't wait to see `core.py` get broken down into more modules. Eventually we should move all of the SQL Lab stuff in `./sqllab.py`

Part of the reason we haven't moved the methods together is that we may have to handle permission migration (they are based on class + method names). I think this is more straigthforward now with somewhat recent improvements in FAB that allow specifying perm names and I think FAB  can do perm migrations in some cases. @dpgaspar

**Pull request merged**